sympy/core/containers.py

```diff
@@ -231,8 +231,9 @@ def __new__(cls, *args):
             items = [Tuple(k, v) for k, v in args]
         else:
             raise TypeError('Pass Dict args as Dict((k1, v1), ...) or Dict({k1: v1, ...})')
+        items = ordered(items)
         elements = frozenset(items)
-        obj = Basic.__new__(cls, elements)
+        obj = Basic.__new__(cls, *items)
         obj.elements = elements
         obj._dict = dict(items)  # In case Tuple decides it wants to sympify
```

Comment:
Since `ordered` returns a one-shot iterable it will already be consumed before we reach this line giving:
```python
In [1]: Dict({x:1})._dict
Out[1]: {}
```